lightningd/bitcoind.c

```diff
@@ -783,6 +783,66 @@ void bitcoind_gettxout(struct bitcoind *bitcoind,
 			  NULL);
 }
 
+static bool extract_numeric_version(struct bitcoin_cli *bcli,
+			    const char *output, size_t output_bytes,
+			    u64 *version)
+{
+	const jsmntok_t *tokens, *versiontok;
+	bool valid;
+
+	tokens = json_parse_input(output, output, output_bytes, &valid);
+	if (!tokens)
+		fatal("%s: %s response",
+		      bcli_args(tmpctx, bcli),
+		      valid ? "partial" : "invalid");
+
+	if (tokens[0].type != JSMN_OBJECT) {
+		log_unusual(bcli->bitcoind->log,
+			    "%s: gave non-object (%.*s)?",
+			    bcli_args(tmpctx, bcli),
+			    (int)output_bytes, output);
+		return false;
+	}
+
+	versiontok = json_get_member(output, tokens, "version");
+	if (!versiontok)
+		return false;
+
+	return json_to_u64(output, versiontok, version);
+}
+
+static bool process_getclientversion(struct bitcoin_cli *bcli)
+{
+	u64 version;
+	u64 min_version = bcli->bitcoind->chainparams->cli_min_supported_version;
+
+	if (!extract_numeric_version(bcli, bcli->output,
+				     bcli->output_bytes,
+				     &version)) {
+		fatal("%s: Unable to getclientversion (%.*s)",
+		      bcli_args(tmpctx, bcli),
+		      (int)bcli->output_bytes,
+		      bcli->output);
+	}
+
+	if (version < min_version)
+		fatal("Unsupported client version? client version: %"PRIu64","
```

Comment:
The word "bitcoind" needs to appear in this message message somewhere?  Otherwise I'd have no idea what client you're referring to!

Reply:
Yes! Corrected.